roles/debian/mattermost/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+---
+- name: Create a new database for mattermost.
+  community.postgresql.postgresql_db:
+    name: mattermost
+    comment: Mattermost Database
+
+- name: Connect to mattermost database, create mattermost user, and grant access to database.
+  community.postgresql.postgresql_user:
+    db: mattermost
+    name: mmuser
+    password: "{{ _encrypted_mattermost_postgresql_pass }}"
+
+- name: GRANT ALL PRIVILEGES ON DATABASE mattermost TO mattermost
+  community.postgresql.postgresql_privs:
+    db: postgres
+    privs: ALL
+    type: database
+    obj: mattermost
+    role: mmuser
+
+- name: Reload PostgreSQL.
+  ansible.builtin.command: /usr/bin/systemctl reload postgresql
+  become: true
+  become_user: root
+
+- name: Check if we already have Mattermost installed.
+  ansible.builtin.stat:
+    path: "/opt/mattermost/bin/mattermost"
+  register: mattermost_bin
+
+- name: Check if Mattermost data directory exists.
+  ansible.builtin.stat:
+    path: "/opt/mattermost/data"
+  register: mattermost_data
+
+- name: "Fetch and extract Mattermost team version {{ mattermost.version }}."
+  ansible.builtin.unarchive:
+    src: "https://releases.mattermost.com/{{ mattermost.version }}/mattermost-team-{{ mattermost.version }}-linux-amd64.tar.gz"
+    dest: /opt
+    remote_src: true
+  when:
+    - not mattermost_bin.stat.exists
+
+- name: Create Mattermost data directory.
+  ansible.builtin.file:
+    path: /opt/mattermost/data
+    state: directory
+    mode: '0755'
+  when:
+    - not mattermost_data.stat.exists
+
+- name: Create mattermost user.
+  ansible.builtin.user:
+    name: mattermost
+    group: mattermost
+    system: true
+
+- name: Recursively set ownership of mattermost directory.
+  ansible.builtin.file:
+    path: /opt/mattermost
+    state: directory
+    recurse: true
+    owner: mattermost
+    group: mattermost
+    mode: g+w
+
+- name: Copy Mattermost systemd service def.
+  ansible.builtin.template:
+    src: "mattermost_service.j2"
```

Comment:
You'll need to provide a default `templates/mattermost_service.j2` file with the role.

Reply:
default systemd unit file template provided for the mattermost role here;

https://github.com/codeenigma/ce-provision/pull/1788/commits/eb6bfb0b16375878583c480f4bb86b60480a8cea